### PR TITLE
refactor icon button with theme props

### PR DESCRIPTION
**Pull request by serdec**

### CODE REVIEW

File: src/components/IconButton/IconButton.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    );
+
+  it("should render the button", () => {
+    const { getAllByRole } = renderButton({ icon: "user" });
+    expect(getAllByRole("button").length).toEqual(1);
+  });
+
+  it("should execute action on click", () => {
+    let counter = 0;
+    const hanldeClick = () => (counter = 1);
```
Comment (fneves) [inline src/components/IconButton/IconButton.test.tsx:21]: typo

Comment (serdec) [reply src/components/IconButton/IconButton.test.tsx:21]: fixed

File: src/components/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -1,94 +1,70 @@
...
 import { HTMLAttributes, forwardRef } from "react";
 import styled from "styled-components";
 
-interface IconButtonProps extends HTMLAttributes<HTMLButtonElement> {
-  state?: "default" | "active";
-  size?: "small" | "default";
+export interface IconButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  size?: "small" | "medium" | "large";
   disabled?: boolean;
   display?: "empty" | "filled";
```
[the remark this answers is not in the document]
Comment (serdec) [reply src/components/IconButton/IconButton.tsx:9]: @gjones we could call this `filled` (true or false). But i have noticed that this matches with `primary` and `secondary`, is that correct? `filled = primary` and `empty = secondary`?

**Comment by vineethasok** [comment]
@fneves Should we merge in this PR or make the changes regarding the naming and then merge it?

**Comment by fneves** [comment]
is there an urgent reason or can we wait for @serdec ?

File: src/components/IconButton/IconButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gjones) [reply src/components/IconButton/IconButton.tsx:9]: @serdec yes, thats correct.

**Pull request merged**